lib/matplotlib/tests/test_lines.py

```diff
@@ -85,22 +85,6 @@ def test_set_line_coll_dash():
     ax.contour(np.random.randn(20, 30), linestyles=[(0, (3, 3))])
 
 
-def test_invalid_line_data():
-    with pytest.raises(RuntimeError, match='xdata must be'):
-        mlines.Line2D(0, [])
-    with pytest.raises(RuntimeError, match='ydata must be'):
-        mlines.Line2D([], 1)
-
-    line = mlines.Line2D([], [])
-    # when deprecation cycle is completed
-    # with pytest.raises(RuntimeError, match='x must be'):
-    with pytest.warns(mpl.MatplotlibDeprecationWarning):
```

Comment:
and we should change these asserts that the calls warn to asserts that they raise (as you changed the warnings into exceptions in the library code).